artman/cli/main2.py

```diff
@@ -74,6 +76,36 @@ def main(*args):
         _run_artman_in_docker(flags)
 
 
+def _adjust_input_dir(input_dir):
+    # Adjust input directory to use versioned shared config under
+    # /googleapis/gapic/[core,lang,packaging]. Currently, the codegen has
+    # coupling with some shared configuration yaml under those directories,
+    # causing library generation to fail when a breaking change is made to such
+    # shared configuration file. This delivers a poor user experience to artman
+    # users, as their library generation could fail without any change at API
+    # proto side.
+    # TODO(ethanbao): Remove this logic once
+    # https://github.com/googleapis/toolkit/issues/1450 is fixed.
+    if os.getenv(RUNNING_IN_ARTMAN_DOCKER_TOKEN):
+        # Only doing this when running inside Docker container
+        DIRS = [
+            'gapic/core',
+            'gapic/lang',
+            'gapic/packaging'
+        ]
+        for src_dir in DIRS:
+            # /googleapis is the root of the versioned googleapis repo
+            # inside Artman Docker image.
+            abs_src_dir = os.path.join('/googleapis', src_dir)
+            for root, dirs, files in os.walk(abs_src_dir):
```

Comment:
Don't you only want to copy `googleapis/gapic`? IIUC with the current setup you'll override any updates to the protos that the user has made if they're running in docker.

Reply:
The src_dir contains "gapic" in the path (see DIRS variable). Technically, I only need three subdirectories under gapic (i.e. core, lang and packaging).

Reply:
Ack, sorry, misread the code.